src/MultiPartStreamFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Http\Message;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Factory for creating Multipart Streams.
+ *
+ * @author Robert Schönthal <robert.schoenthal@gmail.com>
+ */
+interface MultiPartStreamFactory
+{
+    /**
+     * Creates a Multipart Stream from parameters and/or files.
+     *
+     * @param array $parameters
+     * @param array $files
+     *
+     * @return StreamInterface
+     *
+     * @throws \InvalidArgumentException If the $parameters or $files are invalid.
```

Comment:
i could imagine a not readable file, or a something completely different (like e.g. a doctrine Entity or something nonsense...) ?

i used https://github.com/webmozart/assert in another project for input validation